adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -351,6 +389,23 @@ private static void extractJsonObjects(Map<String, String> responseItems, String
         }
     }
 
+    static void extractJsonArray(Map<String, List<String>> responseList, String jsonStr)
```

Comment:
`HashMapExtensions.java` contains _very_ similar functionality to this already. I'd take a look to see if the current implementation does everything you need already. May require some tweaking of the API

Reply:
good catch. thanks